File: specification/specs/indices/index_management/split_index/SplitIndexRequest.ts
@rest_spec_name("indices.split")
class SplitIndexRequest extends RequestBase {
	query_parameters: {
		master_timeout: Time;
		timeout: Time;
		wait_for_active_shards: string;
	}
	body: {
		aliases: Dictionary<IndexName, Alias>;
		settings: Dictionary<string, any>;
	}
}

File: output/csharp/Indices/IndexManagement/SplitIndex/SplitIndexRequest.cs
using Nest.CommonOptions;
using Nest.CommonAbstractions;
using Nest.Internal;
using Nest.Indices;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.Indices {

	public class SplitIndexRequest : RequestBase {
		
		[DataMember(Name="master_timeout")]
		public Time MasterTimeout { get; set; }


		[DataMember(Name="timeout")]
		public Time Timeout { get; set; }


		[DataMember(Name="wait_for_active_shards")]
		public string WaitForActiveShards { get; set; }


		[DataMember(Name="aliases")]
		public IDictionary<IndexName, Alias> Aliases { get; set; }


		[DataMember(Name="settings")]
		public IDictionary<string, object> Settings { get; set; }

	}
}
